python/cuml/preprocessing/model_selection.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cudf
+from typing import Union, Tuple
+from numba import jit
+import numpy as np
+
+
+@jit
+def _shuffle_idx(df: cudf.DataFrame) -> np.ndarray:
+    """ Return indices which represent a randomly shuffled version of `df`
+    """
+    return np.random.permutation(len(df))
+
+
+def train_test_split(
+    X: cudf.DataFrame,
+    y: Union[str, cudf.Series],
+    train_size: float = 0.8,
+    shuffle: bool = True,
```

Comment:
Users will likely want the option to set a seed for their splits for repeatable experiments. Since numpy is used for permutation, you could use numpy to optionally set the random seed.

Additionally, another question is what the function arguments of these kinds of preprocessing methods should look like for eventual, smooth use with dask-ml. @dantegd , @cjnolet

Reply:
Not sure about preprocessing in general, but in this case at least dask-ml's `train_test_split` has the same interface as sklearn's and therefore (to a lesser extent) ours.